openquake/hazardlib/contexts.py

```diff
@@ -548,7 +549,13 @@ def get_mean_stds(self, ctxs):
         :returns: an array of shape (4, G, M, N) with mean and stddevs
         """
         if not hasattr(self, 'imts'):
-            self.imts = tuple(imt_module.from_string(im) for im in self.imtls)
+            tmp = []
+            for im in self.imtls:
+                m = re.match(imt_module.EAS_PERIOD_PATTERN, im)
+                if m:
+                    im = 'EAS({:.6f})'.format(1./float(m.group(1)))
+                tmp.append(imt_module.from_string(im))
+            self.imts = tuple(tmp)
```

Comment:
This logic must go inside `imt.from_string`, not here

Reply:
No, it cannot go there since here the code will try to create an EAS using the period rather than frequency.